### PR TITLE
feat: export upgrades, ezkl_lib upgrades, more tests

### PR DESCRIPTION
**Pull request by JSeam2**
Changes:
1. Upgrades `ezkl_lib` version to `^0.3.0` which has the calibration functionality and forward to gen_witness refactor. **This will result in a breaking change**
2. Bumps minimum python version required to `>=3.8` as `3.7` is starting to be deprecated by various datascience libraries like `numpy`, `scikit-learn`. **This will also result in a breaking change**
3. Includes the new calibration functionality as part of `ezkl`
4. Improves `export` by making `gen-witness` (previously `forward`) optional as well as allowing parameters to be passed within `gen-witness`, makes `calibrate_settings` available within `export` and makes it optional
5. Introduces `model.eval()` prior to export to set model to inference mode. This is important since operators like dropout or batchnorm behave differently in inference and training mode. See https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html
6. Refactors export examples and puts them into a separate folder
7. Initial integration tests
8. ezkl_demo.ipynb updates
9. Updates README.md with clearer quickstart instructions

### CODE REVIEW

File: ezkl/export.py [hunk 1]
```diff
@@ -4,8 +4,21 @@
...
+    torch_model,
+    input_shape=None,
+    input_array=None,
+    onnx_filename="network.onnx",
+    input_filename="input.json",
+    settings_filename="settings.json",
+    run_forward=True,
+    run_calibrate_settings=True,
+    calibration_target="resources",
+    scale=7,
```
Comment (alexander-camuto) [inline ezkl/export.py:19]: set to None by default

**Review by alexander-camuto — APPROVED** [review]
LGTM !
